class-tgm-plugin-activation.php

```diff
@@ -249,7 +249,7 @@ public function init() {
 			// After this point, the plugins should be registered and the configuration set.
 
 			// Proceed only if we have plugins to handle.
-			if ( $this->plugins ) {
+			if ( ! empty( $this->plugins ) ) {
 				$sorted = array();
```

Comment:
1. and 2. - I'm not asking why the need for `is_array()`, I'm asking why you're checking against exactly `array()` instead of using `empty()`. By the time we get to that check, we've already established we're looking at it being an array, so 2. is irrelevant.

[Performance wise](http://maettig.com/code/php/php-performance-benchmarks.php#empty), `empty()` seems to be quicker than `=== array()`.

`empty()` is also what Scrutinizer recommended so I would expect they've done their homework.

`empty()` is also slightly more self-documenting than `=== array()`.

I'm happy to add the `is_array()` check, but the `empty()` should be used over `=== array()` IMO i.e. `if ( is_array( $this->plugins ) && ! empty( $this->plugins ) )`